### PR TITLE
Pool: Status management updates

### PR DESCRIPTION
**Pull request by markuspluna**
Updated status management in accordance with issue #150 and #151

1. Auction Cancellations now have the same action requirements as borrowing (issue #151 )
2. Admin Status behavior changed - changes detailed here (issue #150 )

### CODE REVIEW

File: pool/src/pool/status.rs [hunk 1]
```diff
@@ -1,59 +1,93 @@
...
+                || !met_threshold
+                || (pool_config.status != 0 && pool_backstop_data.q4w_pct >= 0_3000000))
+        {
+            // On-Ice
+            pool_config.status = 3;
+        // Admin status' all supersede backstop triggered active
+        } else if pool_config.status % 2 != 0 {
+            // Active
+            pool_config.status = 1;
+        }
```
Comment (mootz12) [inline pool/src/pool/status.rs:84]: would it be easier to frame this as a switch? I have a hard time following the if statements.

```rust
match cur_status {
    4 -> panic
    2 -> check if we can set to 5
    0 -> check if we can set to 3
    else ->    if pool_backstop_data.q4w_pct >= 0_5000000 {
        pool_config.status = 5;
    } else if pool_backstop_data.q4w_pct >= 0_2500000
        || calc_pool_backstop_threshold(&pool_backstop_data) < SCALAR_7
    {
        pool_config.status = 3;
    } else {
        pool_config.status = 1;
}
```

at the very least, it might be worth trying to make the code a bit more follow-able. If it doesn't, I am cool with this.

File: pool/src/pool/status.rs [hunk 2]
```diff
@@ -378,11 +782,103 @@ mod tests {
...
+        let pool_config = PoolConfig {
+            oracle: oracle_id,
+            bstop_rate: 0,
+            status: 4,
+        };
+        e.as_contract(&pool_id, || {
+            storage::set_admin(&e, &bombadil);
+            storage::set_pool_config(&e, &pool_config);
+
+            execute_update_pool_status(&e, 11);
```
Comment (mootz12) [inline pool/src/pool/status.rs:872]: `11`? I think this was the original reason I split these into two functions, because setting a value here for the permissionless version is very confusing.

Most of the logic can be shared, anyway (fns like `calc_threshold` already share most of the code).

I think I'd prefer the admin and permissionless fn's remain separate

**Review by mootz12 — CHANGES_REQUESTED** [review]

File: pool/src/pool/pool.rs [hunk 3]
```diff
@@ -296,6 +296,45 @@ mod tests {
...
+    #[should_panic(expected = "Error(Contract, #11)")]
+    fn test_require_action_allowed_cancel_liquidation_while_on_ice_panics() {
+        let e = Env::default();
+
+        let pool = testutils::create_pool(&e);
+        let oracle = Address::generate(&e);
+        let pool_config = PoolConfig {
+            oracle,
+            bstop_rate: 0_200_000_000,
+            status: 3,
```
[the remark this answers is not in the document]
Comment (markuspluna) [reply pool/src/pool/pool.rs:309]: Also updated other tests to check edge status'

File: pool/src/pool/status.rs (hunk 1 above)
Comment (markuspluna) [reply pool/src/pool/status.rs:84]: match statement helped

**Review by mootz12 — APPROVED** [review]
